### PR TITLE
Separate Issuer, Verifier, Holder APIs.

### PR DESCRIPTION
**Pull request by peacekeeper**
This separates the VC HTTP API into three separate files for an Issuer API, Verifier API, and Holder API. It also separates out some common schemas (such as "VerifiableCredential") so that they can be re-used independently.

I don't feel too strongly about this, so I'm also fine if this doesn't get merged. But maybe this can be a contribution to the ongoing thought process on how to organize and document things better.

### CODE REVIEW

**Review by OR13 — APPROVED** [review]

**Review by mprorock — APPROVED** [review]
this is a great step in the right direction

**Comment by peacekeeper** [comment]
This PR was discussed in a [VC HTTP API Special Topic Call on 2021-04-22](https://w3c-ccg.github.io/meetings/2021-04-22-vchttpapi/).

File: docs/components/Credential.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        "id":
+          type: string
+          description: The ID of the credential.
+        "type":
+          type: array
+          description: The JSON-LD type of the credential.
+          items:
+            type: string
+        "issuer":
+          $ref: "#/components/schemas/Issuer"
```
Comment (brianorwhatever) [inline docs/components/Credential.yml:33]: ```./Issuer.yml#/components/schemas/Issuer```

File: docs/components/VerifiableCredential.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    name: GitHub Source Code
+    url: https://github.com/w3c-ccg/vc-http-api
+paths:
+components:
+  schemas:
+    VerifiableCredential:
+      type: object
+      description: A JSON-LD Verifiable Credential with a proof.
+      allOf:
+        - $ref: "#/components/schemas/Credential"
```
Comment (brianorwhatever) [inline docs/components/VerifiableCredential.yml:19]: ```./Credential.yml#/components/schemas/Credential```

File: docs/components/VerifiableCredential.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        - type: object
+          properties:
+            proof:
+              $ref: "#/components/schemas/LinkedDataProof"
```
Comment (brianorwhatever) [inline docs/components/VerifiableCredential.yml:23]: ```./LinkedDataProof.yml#/components/schemas/LinkedDataProof```

File: docs/components/VerifiablePresentation.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    name: GitHub Source Code
+    url: https://github.com/w3c-ccg/vc-http-api
+paths:
+components:
+  schemas:
+    VerifiablePresentation:
+      type: object
+      description: A JSON-LD Verifiable Presentation with a proof.
+      allOf:
+        - $ref: "#/components/schemas/Presentation"
```
Comment (brianorwhatever) [inline docs/components/VerifiablePresentation.yml:19]: ```./Presentation.yml#/components/schemas/Presentation```

File: docs/components/VerifiablePresentation.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        - type: object
+          properties:
+            proof:
+              $ref: "#/components/schemas/LinkedDataProof"
```
Comment (brianorwhatever) [inline docs/components/VerifiablePresentation.yml:23]: ```./LinkedDataProof.yml#/components/schemas/LinkedDataProof ```

**Review by brianorwhatever — COMMENTED** [review]
some external ref links missing

**Comment by peacekeeper** [comment]
> some external ref links missing

Thanks! Just fixed those in https://github.com/w3c-ccg/vc-http-api/pull/178/commits/87dcd79b5b35571030400e8d8c38946200324a7e.

**Comment by msporny** [comment]
@peacekeeper can you take this PR out of draft... We now have a resolution where we can pull it into the repo:

https://w3c-ccg.github.io/meetings/2021-05-06-vchttpapi/#resolution-1

**Comment by peacekeeper** [comment]
> can you take this PR out of draft

done

**Pull request merged**